godo.go

```diff
@@ -271,6 +278,16 @@ func New(httpClient *http.Client, opts ...ClientOpt) (*Client, error) {
 		}
 	}
 
+	// if retryMax is set it will use the retryablehttp client.
+	if c.retryMax > 0 {
+		retryableClient := retryablehttp.NewClient()
+		retryableClient.RetryMax = c.retryMax
+		retryableClient.RetryWaitMin = time.Duration(c.retryWaitMin * float64(time.Second))
+		retryableClient.RetryWaitMax = time.Duration(c.retryWaitMin * float64(time.Second))
+
+		c.client = retryableClient.StandardClient()
```

Comment:
Unfortunately, since this replaces the provided client, this clobbers the OAuth token. If you build a little program using this version of godo, you'll always get a 401.

There are ways to work around this. For example:

```suggestion
		var source *oauth2.Transport
		if _, ok := c.client.Transport.(*oauth2.Transport); ok {
			source = c.client.Transport.(*oauth2.Transport)
		}

		c.client = retryableClient.StandardClient()
		c.client.Transport = &oauth2.Transport{
			Base:   c.client.Transport,
			Source: source.Source,
		}
```

If we document it, it is probably fine since it is optional. Though I wonder if this is too cute? We allow users to pass in any `*http.Client`. Are there other things we might be overwriting? Another option is to create another method for instantiation a client, e.g. `NewRetryable()`. Though I'm not sure I love that either, we already have `New()`, `NewClient()`, and `NewFromToken()`. It's getting a bit difficult to navigate...